services/requester/requester.go

```diff
@@ -597,7 +526,111 @@ func (e *EVM) dryRunTx(
 	return result, nil
 }
 
-func AddOne64th(n uint64) uint64 {
-	// NOTE: Go's integer division floors, but that is desirable here
-	return n + (n / 64)
+// buildTransaction creates a flow transaction from the provided script,
+// with the arguments and signs it with the configured COA account.
+func (e *EVM) buildTransaction(
+	ctx context.Context,
+	script []byte,
+	args ...cadence.Value,
+) (*flow.Transaction, error) {
+	// building and signing transactions should be blocking, so we don't have keys conflict
+	e.mux.Lock()
+	defer e.mux.Unlock()
+
+	var (
+		g           = errgroup.Group{}
+		err1, err2  error
+		latestBlock *flow.Block
+		account     *flow.Account
+	)
+	// execute concurrently so we can speed up all the information we need for tx
+	g.Go(func() error {
+		latestBlock, err1 = e.client.GetLatestBlock(ctx, true)
+		return err1
+	})
+	g.Go(func() error {
+		account, err2 = e.client.GetAccount(ctx, e.config.COAAddress)
+		return err2
+	})
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+
+	accKey, err := e.keystore.GetKey()
+	if err != nil {
+		return nil, err
+	}
+
+	flowTx := flow.NewTransaction().
+		SetScript(script).
+		SetReferenceBlockID(latestBlock.ID)
+
+	for _, arg := range args {
+		if err := flowTx.AddArgument(arg); err != nil {
+			return nil, fmt.Errorf("failed to add argument: %s, with %w", arg, err)
+		}
+	}
+
+	if err := accKey.SetProposerPayerAndSign(flowTx); err != nil {
+		return nil, err
+	}
+	if err := accKey.IncrementSequenceNumber(); err != nil {
+		return nil, err
+	}
+	e.keystore.LockKey(flowTx.ID(), accKey)
+
+	e.collector.OperatorBalance(account)
+
+	return flowTx, nil
+}
+
+// validateTransactionWithState checks if the given tx has the correct
+// nonce & balance, according to the local state.
+func (e *EVM) validateTransactionWithState(
+	tx *types.Transaction,
+	from common.Address,
+) error {
+	height, err := e.blocks.LatestEVMHeight()
+	if err != nil {
+		return err
+	}
+	view, err := e.getBlockView(height)
+	if err != nil {
+		return err
+	}
+
+	nonce, err := view.GetNonce(from)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the transaction adheres to nonce ordering
```

Comment:
**Note:** We only check for `nonce too low`, to match the functionality from Geth:
- https://github.com/onflow/go-ethereum/blob/master/core/txpool/validation.go#L209-L219
- https://github.com/onflow/go-ethereum/blob/master/core/txpool/legacypool/legacypool.go#L631

This has the benefit that it allows users to sign and submit sequential transactions from an EOA. It is the sender's responsibility to make sure the correct nonce was used.

In EVM Gateway, if we checked for `nonce too high`, then we produce false negatives, as the transactions might be in flight, and the local index has not yet caught up.